arangod/Replication2/ReplicatedLog/LogCommon.cpp

```diff
@@ -0,0 +1,305 @@
+  return value <= other.value;
+}
+
+LogTerm::operator velocypack::Value() const noexcept {
+  return velocypack::Value(value);
+}
+
+auto replication2::operator<<(std::ostream& os, const LogTerm& term) -> std::ostream& {
+  return os << term.value;
+}
+
+auto replication2::operator==(LogPayload const& left, LogPayload const& right) -> bool {
+  return arangodb::basics::VelocyPackHelper::equal(velocypack::Slice(left.dummy.data()),
+                                                   velocypack::Slice(right.dummy.data()),
+                                                   true);
+}
+
+auto replication2::operator!=(LogPayload const& left, LogPayload const& right) -> bool {
+  return !(left == right);
+}
+
+LogPayload::LogPayload(velocypack::UInt8Buffer dummy)
+    : dummy(std::move(dummy)) {}
+
+auto LogPayload::createFromSlice(velocypack::Slice slice) -> LogPayload {
+  VPackBufferUInt8 buffer;
+  VPackBuilder builder(buffer);
+  builder.add(slice);
+  return LogPayload(std::move(buffer));
+}
+
+auto LogPayload::createFromString(std::string_view string) -> LogPayload {
+  VPackBufferUInt8 buffer;
+  VPackBuilder builder(buffer);
+  builder.add(VPackValue(string));
+  return LogPayload(std::move(buffer));
+}
+
+auto LogPayload::byteSize() const noexcept -> std::size_t {
+  return dummy.byteSize();
+}
+
+auto LogPayload::slice() const noexcept -> velocypack::Slice {
+  return VPackSlice(dummy.data());
+}
+
+auto LogId::fromString(std::string_view name) noexcept -> std::optional<LogId> {
+  if (std::all_of(name.begin(), name.end(), [](char c) { return isdigit(c); })) {
+    using namespace basics::StringUtils;
+    return LogId{uint64(name)};
+  }
+  return std::nullopt;
+}
+
+[[nodiscard]] LogId::operator velocypack::Value() const noexcept {
+  return velocypack::Value(id());
+}
+
+auto replication2::to_string(LogId logId) -> std::string {
+  return std::to_string(logId.id());
+}
+
+auto replication2::to_string(LogTerm term) -> std::string {
+  return std::to_string(term.value);
+}
+
+auto replication2::to_string(LogIndex index) -> std::string {
+  return std::to_string(index.value);
+}
+
+auto replication2::operator<=(replication2::TermIndexPair const& left,
+                              replication2::TermIndexPair const& right) noexcept -> bool {
+  if (left.term < right.term) {
+    return true;
+  } else if (left.term == right.term) {
+    return left.index <= right.index;
+  } else {
+    return false;
+  }
+}
+
+void replication2::TermIndexPair::toVelocyPack(velocypack::Builder& builder) const {
+  VPackObjectBuilder ob(&builder);
+  builder.add(StaticStrings::Term, VPackValue(term.value));
+  builder.add(StaticStrings::Index, VPackValue(index.value));
+}
+
+auto replication2::TermIndexPair::fromVelocyPack(velocypack::Slice slice) -> TermIndexPair {
+  TermIndexPair pair;
+  pair.term = LogTerm{slice.get(StaticStrings::Term).getNumericValue<size_t>()};
+  pair.index = LogIndex{slice.get(StaticStrings::Index).getNumericValue<size_t>()};
```

Comment:
The types used here should match the internal types (i.e. `value`) although in practice size_t and uint64_t will be equivalent for our target platforms.
```suggestion
  pair.term = LogTerm{slice.get(StaticStrings::Term).getNumericValue<uint64_t>()};
  pair.index = LogIndex{slice.get(StaticStrings::Index).getNumericValue<uint64_t>()};
```

Reply:
I use the extractor instead.